.github/actions/verify-k8s-explorer-logs/action.yml

```diff
@@ -0,0 +1,70 @@
+name: "Verify Kubernetes Explorer Logs"
+description: "Action which verifies that Kubernetes and Kubernetes Explorer logs have been correctly ingested."
+
+inputs:
+  scalyr_readlog_token:
+    description: "Scalyr readlogs API key to use"
+    required: true
+  scalyr_agent_pod_name:
+    description: "Name of the scalyr agent pod."
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+    - name: Verify Agent Logs are Ingested
+      shell: bash
+      env:
+        scalyr_readlog_token: "${{ inputs.scalyr_readlog_token}}"
+        SCALYR_AGENT_POD_NAME: "${{ inputs.scalyr_agent_pod_name }}"
+      run: |
+        # Verify agent and kubernetes monitor has been started
+        ./ci/scripts/scalyr-query.sh '$serverHost="'${SCALYR_AGENT_POD_NAME}'" $logfile="/var/log/scalyr-agent-2/agent.log" "Starting scalyr agent..."'
+        ./ci/scripts/scalyr-query.sh '$serverHost="'${SCALYR_AGENT_POD_NAME}'" $logfile="/var/log/scalyr-agent-2/agent.log" "No checkpoints were found. All logs will be copied starting at their current end"'
+
+        ./ci/scripts/scalyr-query.sh '$serverHost="'${SCALYR_AGENT_POD_NAME}'" $logfile="/var/log/scalyr-agent-2/agent.log" "Cluster name detected, enabling k8s metric reporting"'
+        ./ci/scripts/scalyr-query.sh '$serverHost="'${SCALYR_AGENT_POD_NAME}'" $logfile="/var/log/scalyr-agent-2/agent.log" "kubernetes_monitor parameters: "'
+
+        #./ci/scripts/scalyr-query.sh '$serverHost="'${SCALYR_AGENT_POD_NAME}'" $logfile="/var/log/scalyr-agent-2/agent.log" "Config option '\'stop_agent_on_failure\'' is enabled"'
```

Comment:
I guess we can uncomment this since new stable release has to include that.

Reply:
We do actually check for it in a (simpler) query below.

> ./ci/scripts/scalyr-query.sh '$serverHost="'${SCALYR_AGENT_POD_NAME}'" $logfile="/var/log/scalyr-agent-2/agent.log" "stop_agent_on_failure"'

I disabled this query since I had some problems with getting it work (shell param escaping - I tried various escape sequences, but it didn't work).

That's one of the downsides of this simple bash based query script. Perhaps in the future it would make sense to add support for specifying query filters using JSON object.

Reply:
Got it.